src/main/frontend/src/components/landingpage/InfiniteScrollComponent.js

```diff
@@ -0,0 +1,74 @@
+import "bootstrap/dist/css/bootstrap.min.css";
+import InfiniteScroll from "react-infinite-scroll-component";
+import { useEffect, useState } from "react";
+import LandingPageItem from "./LandingPageItem";
+import { Row } from "react-bootstrap";
+import {itemsFetch, itemsFetchByDate} from './itemService';
+
+function InfiniteScrollComponent(props) {
+  const [items, setItems] = useState([]);
+
+  const [hasMoreItems, setHasMoreItems] = useState(true);
+
+  const [page, setpage] = useState(0);
+
+  useEffect(async () => {
+    let data = "";
+
+    if(props.criteria === "normal"){
+      data = await itemsFetch(page, 20);
+    }else if(props.criteria === "last_chance"){
+      data = await itemsFetchByDate(page, 20, "desc");
+    }
+
+    setItems(data.content);
+  }, []);
+
+  const fetchData = async () => {
+    let commentsFromServer = "";
+
+    if(props.criteria === "normal"){
+      commentsFromServer = await itemsFetch(page, 20);
+    }else if(props.criteria === "last_chance"){
+      commentsFromServer = await itemsFetchByDate(page, 20, "desc");
+    }
+
+    setItems([...items, ...commentsFromServer.content]);
+
+    setHasMoreItems(commentsFromServer.last);
```

Comment:
We can extract this to separate useEffect that will have page in its dependency-array. That way first useEffect can be deleted. And only thing that can stay in fetch data will be setPage